plugins/woocommerce/tests/php/includes/class-wc-ajax-test.php

```
@@ -822,6 +822,70 @@ public function test_refund_line_items_amount_only_skips_untouched_items(): void
+		$coupon->save();
+
+		$_POST['security'] = wp_create_nonce( 'remove-coupon' );
+		$_POST['coupon']   = $coupon->get_code();
+
+		// Without a filter attached, the notice is the get_coupon_message() text for WC_COUPON_REMOVED.
+		$default_response = $this->handle_remove_coupon_ajax();
+		$this->assertStringContainsString( 'Coupon code removed successfully.', $default_response, 'The notice should use the get_coupon_message() text.' );
+		$this->assertStringNotContainsString( 'Coupon has been removed.', $default_response, 'The old hardcoded text should no longer be used.' );
+
+		$captured_code = null;
+		$filter        = function ( $message, $msg_code ) use ( &$captured_code ) {
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$message'. (undefined)

(UnusedFormalParameter)